CMakeLists.txt

```diff
@@ -6,60 +6,33 @@ set(CMAKE_BUILD_TYPE "Release")
 
 project (ssplit CXX)
 
-# Default is C++17, check if compiler supports, else hard-fail asking user to explicitly mention C++-11
-include(CheckCXXCompilerFlag)
-if(MSVC) # MSVC compiler has different flags
-    CHECK_CXX_COMPILER_FLAG("/std:c++17" COMPILER_SUPPORTS_CXX17)
-    CHECK_CXX_COMPILER_FLAG("/std:c++11" COMPILER_SUPPORTS_CXX11)
-else(MSVC)
-    CHECK_CXX_COMPILER_FLAG("-std=c++17" COMPILER_SUPPORTS_CXX17)
-    CHECK_CXX_COMPILER_FLAG("-std=c++11" COMPILER_SUPPORTS_CXX11)
-endif(MSVC)
-if(NOT COMPILER_SUPPORTS_CXX17 AND NOT USE_CXX11)
-    message(SEND_ERROR "Please enable -DUSE_CXX11=on as compiler does not support C++17." 
-                       "This will enable abseil-cpp library")
-endif()
+option(SSPLIT_USE_INTERNAL_PCRE2 "Use internal PCRE2 instead of system's." OFF)
+option(SSPLIT_COMPILE_LIBRARY_ONLY "Compile library only." OFF)
 
-
-option(SSPLIT_USE_CXX11 "Restrict to C++-11, enables abseil string-view" OFF)
-option(SSPLIT_USE_INTERNAL_PCRE2 "Use internal PCRE2 instead of system PCRE2" OFF)
-option(SSPLIT_COMPILE_LIBRARY_ONLY "Compile library only" OFF)
-
-# Make SSPLIT_ prefixed option local.
-set(USE_CXX11 ${SSPLIT_USE_CXX11})
-set(USE_INTERNAL_PCRE2 ${SSPLIT_USE_INTERNAL_PCRE2})
-set(COMPILE_LIBRARY_ONLY ${SSPLIT_COMPILE_LIBRARY_ONLY})
-set(USE_ABSEIL OFF) # Default USE_ABSEIL explicitly turned OFF
-set(EXT_LIBS "") # Initialize EXT_LIBS with a legal value.
-
-# Handle C++ Standards
-if (USE_CXX11)
-  set(CMAKE_CXX_STANDARD 11)
-
-  # If CXX11, abseil-cpp has to be enabled and added to EXT_LIBS, since there's
-  # no way to obtain string_view (required) without.
-  set(USE_ABSEIL ON)
-  include(ProvideAbseil)
-  set(EXT_LIBS ${EXT_LIBS} -Wl,--start-group ${ABSEIL_LIBS} -Wl,--end-group)
-
-  # The source files are imported in the library above. If included as a
-  # subdirectory, we would want this definition to stick for future includes as
-  # well. Hence add_definitions here.
-  add_definitions(-DUSE_ABSEIL)
-else(USE_CXX11)
-  set(CMAKE_CXX_STANDARD 17)
-endif(USE_CXX11)
-
-# Enforce the standard set above.
 set(CMAKE_CXX_STANDARD_REQUIRED ON)
+if (${CMAKE_VERSION} VERSION_GREATER "3.8.0")
+  set(CMAKE_CXX_STANDARD 17)
```

Comment:
Why do we sometimes use C++17 and sometimes C++14? We can manually set C++17, by appending `-std=c++17` to CXXFLAGS (and some ifdefs of MSVC) if necessary. My understanding is that if we have C++17, the work with maps would be slightly easier for other people? We have C++17 with everything but the clang on 16.04. Would it make sense to enforce c++17 , or is someone (emscripten) preventing us from doing it?

Reply:
Simplicity, of sorts. In my opinion, there should be a single mechanism to enforce the c++ standard, rather than setting cxx flags explicitly. Using CMAKE_CXX_STANDARD dumps the responsibility of setting flags according to OS / Compiler onto cmake. This is a fringe case anyway that will hopefully go away soon once we upgrade valhalla to an actually supported versoin of Ubuntu. On a marginal note: I've jumped the gun and issued a pull request on browsermt/ssplit-cpp; with a few minor fixes, it's the same as this PR. Jerin has approved the PR; once you approve, I'll merge. Thanks, by the way, for all the work you've put into making this compile on Windows and overall cleanup.